cmd/agent/main.go

```diff
@@ -67,13 +68,15 @@ func main() {
 
 			rOpts := new(reporter.Options).InitFromViper(v, logger)
 			grpcBuilder, err := grpc.NewConnBuilder().InitFromViper(v)
+			builderCtx, cancel := context.WithCancel(context.Background())
+			defer cancel()
```

Comment:
```suggestion
			ctx, cancel := context.WithCancel(context.Background())
			defer cancel()
```
Also, this should be defined right before it's used, in L79.

Reply:
I have included the changes, and thanks to your comment, I added the package_test.go using the new syntax as suggested.